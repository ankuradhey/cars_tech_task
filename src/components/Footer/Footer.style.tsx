import styled from "styled-components";

export const Footer = styled.footer`
    height: 80px;
    display: flex;
    align-items: center;
    justify-content: center;
    border-top: solid 1px ${(props) => props.theme.palette.common.white};
`;
