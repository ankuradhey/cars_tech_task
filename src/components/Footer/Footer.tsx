import React, { FC } from "react";
import * as Styled from "./Footer.style";

export const Footer: FC = () => {
    return (
        <Styled.Footer className="footer text-center mt-3">
            <span className="margin-auto">Â© AUTO1 Group 2018</span>
        </Styled.Footer>
    );
};
