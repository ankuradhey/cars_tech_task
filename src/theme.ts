import { DefaultTheme } from "styled-components";

export const defaultTheme: DefaultTheme = {
    borderRadius: "2px",
    palette: {
        common: {
            black: "#4A4A4A",
            white: "#EDEDED",
        },
        primary: {
            main: "#EA7F28",
            contrastText: "#ffffff",
        },
        secondary: {
            main: "#D37324",
            contrastText: "#ffffff",
        },
        tertiary: {
            main: "#ffffff",
            contrastText: "#EDEDED",
        },
    },
    gutterSize: {
        sm: "8px",
        md: "12px",
        lg: "24px",
    },
};
