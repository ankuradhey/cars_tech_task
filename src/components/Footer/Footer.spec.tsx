import React from "react";
import { render } from "@testing-library/react";
import { ThemeProvider } from "styled-components";
import { defaultTheme } from "../../theme";
import { Footer } from "./Footer";

describe("Footer Component", () => {
    it("Should render Footer component", () => {
        const { container, getByText } = render(
            <ThemeProvider theme={defaultTheme}>
                <Footer />
            </ThemeProvider>
        );
        expect(getByText("Â© AUTO1 Group 2018")).toBeInTheDocument();
        expect(container.firstChild).toMatchSnapshot();
    });
});
