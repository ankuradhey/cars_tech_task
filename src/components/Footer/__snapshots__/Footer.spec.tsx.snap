// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Footer Component Should render Footer component 1`] = `
<footer
  class="sc-bdfBwQ hhKrKQ footer text-center mt-3"
>
  <span
    class="margin-auto"
  >
    Â© AUTO1 Group 2018
  </span>
</footer>
`;
